gammapy/background/models.py

```diff
@@ -71,3 +77,543 @@ def evaluate(self, x, y):
+        else:
+            raise ValueError("Cannot interpret units ({})".format(background_unit))
+        background = Quantity(background, background_unit)
+
+        return CubeBackgroundModel(det_bins=det_bins,
+                                   energy_bins=energy_bins,
+                                   background=background)
+
+    @property
+    def image_extent(self):
+        """Image extent `(x_lo, x_hi, y_lo, y_hi)`.
+
+        Returns
+        -------
+        im_extent : `~astropy.coordinates.Angle`
+            array of bins with the image extent
+        """
+        b = self.det_bins.degree
+        return Angle([b[0], b[-1], b[0], b[-1]], 'degree')
+
+
+    @property
+    def spectrum_extent(self):
+        """Spectrum extent `(e_lo, e_hi)`.
+
+        Returns
+        -------
+        spec_extent : `~astropy.units.Quantity`
+            array of bins with the spectrum extent
+        """
+        b = self.energy_bins.to('TeV')
+        return Quantity([b[0], b[-1]], 'TeV')
+
+
+    @property
+    def image_bin_centers(self):
+        """Image bin centers `(x, y)`.
+
+        Returns
+        -------
+        det_edges_centers : `~astropy.coordinates.Angle`
+            array of bins with the image bin centers
+        """
+        det_edges_low = self.det_bins[:-1]
+        det_edges_high = self.det_bins[1:]
+        det_edges_centers = (det_edges_low + det_edges_high)/2.
+        return Angle([det_edges_centers, det_edges_centers])
+
+
+    @property
+    def spectrum_bin_centers(self):
+        """Spectrum bin centers (log center)
+
+        Returns
+        -------
+        energy_bin_centers : `~astropy.units.Quantity`
+            array of bins with the spectrum bin centers
+        """
+        energy_edges_low = self.energy_bins[:-1]
+        energy_edges_high = self.energy_bins[1:]
+        energy_bin_centers = 10.**(0.5*(np.log10(energy_edges_low.to('TeV').value*energy_edges_high.to('TeV').value)))
+        return Quantity(energy_bin_centers, 'TeV')
+
+
+    def find_det_bin(self, det):
+        """Find the bin that contains the specified det (X, Y) pair.
+
+        Parameters
+        ----------
+        det : `~astropy.coordinates.Angle`
+            det (X, Y) pair to search for
+
+        Returns
+        -------
+        bin_pos : `~int`
+            index of the det bin containing the specified det (X, Y) pair
+        bin_edges : `~astropy.units.Quantity`
+            det bin edges (x_lo, x_hi, y_lo, y_hi)
+        """
+        # check shape of det: only 1 pair is accepted
+        nvalues = len(det.flatten())
+        if nvalues != 2:
+            print("Expected exactly 2 values for det (X, Y), got {}.".format(nvalues))
+            raise IndexError
+
+        # check that the specified det is within the boundaries of the model
+        det_extent = self.image_extent
+        if not ((det_extent[0] <= det[0]) and (det[0] < det_extent[1])) or not ((det_extent[2] <= det[1]) and (det[1] < det_extent[3])):
```

Comment:
Don't write such long lines. Use temp variables.
(look at it on Github ... if you have to scroll to the right to read the code, it's too long.)

Use an editor that shows you a vertical line at ~80 characters and does PEP8 for you as you type.
It really helps to just write clean code from the start, after a day or two you don't even think about it any more.